### PR TITLE
[Rule Tuning] Convert to EQL 5 existing rules

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/413

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/credential_access_credential_dumping_msbuild.toml [hunk 1]
```diff
@@ -23,9 +23,9 @@ tags = ["Elastic", "Windows"]
 type = "query"
 
 query = '''
-event.category:process and event.type:change and
+event.category:library and event.type:change and
   (winlog.event_data.OriginalFileName:(vaultcli.dll or SAMLib.DLL) or
-    dll.name:(vaultcli.dll or SAMLib.DLL)) and
+    dll.name:(vaultcli.dll or SAMLib.DLL) or file.name:(vaultcli.dll or SAMLib.DLL)) and
   process.name: MSBuild.exe
```
Comment (brokensound77) [inline rules/windows/credential_access_credential_dumping_msbuild.toml:29]: I don't think this is compatible with winlogbeat and endpoint (may not have been in its original state either).

File: rules/windows/credential_access_iis_apppoolsa_pwd_appcmd.toml [hunk 2]
```diff
@@ -21,12 +21,12 @@ risk_score = 73
...
-type = "query"
+type = "eql"
 
 query = '''
-event.category:process AND event.type:(start OR process_started) AND
-  (process.name:appcmd.exe OR process.pe.original_file_name:appcmd.exe) AND
-  process.args:(/[lL][iI][sS][tT]/ AND /\/[tT][eE][xX][tT]\:[pP][aA][sS][sS][wW][oO][rR][dD]/)
+process where event.type in ("start", "process_started") and
+  (process.name == "appcmd.exe" or process.pe.original_file_name == "appcmd.exe") and
+  wildcard(process.args, "/list") and wildcard(process.args, "/text:password")
```
Comment (brokensound77) [inline rules/windows/credential_access_iis_apppoolsa_pwd_appcmd.toml:29]:
```suggestion
process where event.type in ("start", "process_started") and
  (process.name : "appcmd.exe" or process.pe.original_file_name == "appcmd.exe") and
  process.args : "/list" and process.args : "/text:password"
```

File: rules/windows/credential_access_iis_connectionstrings_dumping.toml [hunk 3]
```diff
@@ -25,12 +25,12 @@ risk_score = 73
...
-type = "query"
+type = "eql"
 
 query = '''
-event.category:process and event.type:(start or process_started) and
-  (process.name:aspnet_regiis.exe or process.pe.original_file_name:aspnet_regiis.exe) and 
-  process.args:(connectionStrings and "-pdf")
+process where event.type in ("start", "process_started") and
+  (process.name == "aspnet_regiis.exe" or process.pe.original_file_name == "aspnet_regiis.exe") and
+  wildcard(process.args, "connectionStrings") and wildcard(process.args, "-pdf")
```
Comment (brokensound77) [inline rules/windows/credential_access_iis_connectionstrings_dumping.toml:33]:
```suggestion
process where event.type in ("start", "process_started") and
  (process.name : "aspnet_regiis.exe" or process.pe.original_file_name == "aspnet_regiis.exe") and
  process.args : "connectionStrings" and process.args : "-pdf"
```

File: rules/windows/defense_evasion_masquerading_renamed_autoit.toml [hunk 4]
```diff
@@ -12,19 +12,18 @@ to avoid detection.
...
 tags = ["Elastic", "Windows"]
-type = "query"
+type = "eql"
 
 query = '''
-event.category:process AND event.type:(start OR process_started) AND
-  process.pe.original_file_name:/[aA][uU][tT][oO][iI][tT]\d\.[eE][xX][eE]/ AND
-  NOT process.name:/[aA][uU][tT][oO][iI][tT]\d{1,3}\.[eE][xX][eE]/
+process where event.type in ("start", "process_started", "info") and
+  wildcard(process.pe.original_file_name,"AutoIt*.exe") and not wildcard(process.name,"AutoIt*.exe")
```
Comment (brokensound77) [inline rules/windows/defense_evasion_masquerading_renamed_autoit.toml:26]:
```suggestion
process where event.type in ("start", "process_started", "info") and
  process.pe.original_file_name : "AutoIt*.exe" and not process.name : "AutoIt*.exe"
```

File: rules/windows/defense_evasion_suspicious_zoom_child_process.toml [hunk 5]
```diff
@@ -10,29 +10,20 @@ description = """
...
- not process.name:(Zoom.exe or WerFault.exe or airhost.exe or CptControl.exe or CptHost.exe or cpthost.exe or
-   CptInstall.exe or CptService.exe or Installer.exe or zCrashReport.exe or Zoom_launcher.exe or zTscoder.exe or
-   plugin_Launcher.exe or mDNSResponder.exe or zDevHelper.exe or APcptControl.exe or CrashSender*.exe or aomhost64.exe or
-   Magnify.exe or m_plugin_launcher.exe or com.zoom.us.zTranscode.exe or RoomConnector.exe or tabtip.exe or Explorer.exe or
-   chrome.exe or firefox.exe or iexplore.exe or outlook.exe or lync.exe or ApplicationFrameHost.exe or ZoomAirhostInstaller.exe or
-   narrator.exe or NVDA.exe or Magnify.exe or Outlook.exe or m_plugin_launcher.exe or mphost.exe or APcptControl.exe or winword.exe or
-   excel.exe or powerpnt.exe or ONENOTE.EXE or wpp.exe or debug_message.exe or zAssistant.exe or msiexec.exe or msedge.exe or
-   dwm.exe or vcredist_x86.exe or Controller.exe or Installer.exe or CptInstall.exe or Zoom_launcher.exe or ShellExperienceHost.exe or wps.exe)
+process where event.type in ("start", "process_started", "info") and
+ process.parent.name == "Zoom.exe" and process.name in ("cmd.exe", "powershell.exe")
```
Comment (brokensound77) [inline rules/windows/defense_evasion_suspicious_zoom_child_process.toml:26]:
```suggestion
process where event.type in ("start", "process_started", "info") and
 process.parent.name : "Zoom.exe" and process.name : ("cmd.exe", "powershell.exe")
```

File: rules/windows/evasion_rundll32_no_arguments.toml [hunk 6]
```diff
@@ -31,9 +31,7 @@ sequence with maxspan=1h
       (process.args != "rundll32.exe" and process.args_count == 0))
 
   ] by process.entity_id
-  [process where event.type in ("start", "process_started") and
-     (process.name == "rundll32.exe" or process.pe.original_file_name == "rundll32.exe")
-  ] by process.parent.entity_id
+  [process where event.type in ("start", "process_started") and process.parent.name == "rundll32.exe"] by process.parent.entity_id
```
Comment (brokensound77) [inline rules/windows/evasion_rundll32_no_arguments.toml:34]:
```suggestion
  [process where event.type in ("start", "process_started") and process.parent.name : "rundll32.exe"] by process.parent.entity_id
```

File: rules/windows/credential_access_credential_dumping_msbuild.toml (hunk 1 above)
Comment (Samirbous) [reply rules/windows/credential_access_credential_dumping_msbuild.toml:29]: agree, will delete this change from this PR, need to be reworked.

**Comment by Samirbous** [comment]
@brokensound77 same error as other failing PRs 

image

**Comment by brokensound77** [comment]
@Samirbous the error is because you need to merge changes from main, to pull the latest pyEQL which understands the latest syntax changes. It cannot pull until you resolve those conflicts

**Comment by rw-access** [comment]
@brokensound77 it's using the latest changes. But I'm not sure why it's failing.

File: rules/windows/credential_access_iis_apppoolsa_pwd_appcmd.toml [hunk 7]
```diff
@@ -21,12 +21,12 @@ risk_score = 73
...
 tags = ["Elastic", "Windows"]
-type = "query"
+type = "eql"
 
 query = '''
-event.category:process AND event.type:(start OR process_started) AND
-  (process.name:appcmd.exe OR process.pe.original_file_name:appcmd.exe) AND
-  process.args:(/[lL][iI][sS][tT]/ AND /\/[tT][eE][xX][tT]\:[pP][aA][sS][sS][wW][oO][rR][dD]/)
+process where event.type in ("start", "process_started") and
+   (process.name : "appcmd.exe" or process.pe.original_file_name : "appcmd.exe") and
```
Comment (brokensound77) [inline rules/windows/credential_access_iis_apppoolsa_pwd_appcmd.toml:28]: original_file_name compares _should_ stay case sensitive

```suggestion
   (process.name : "appcmd.exe" or process.pe.original_file_name == "appcmd.exe") and 
```

File: rules/windows/defense_evasion_masquerading_renamed_autoit.toml [hunk 8]
```diff
@@ -1,31 +1,28 @@
...
 creation_date = "2020/09/01"
-ecs_version = ["1.6.0"]
 maturity = "production"
 updated_date = "2020/11/03"
 
 [rule]
 author = ["Elastic"]
 description = """
-Identifies a suspicious AutoIt process execution. Malware written as AutoIt scripts tend to rename the AutoIt executable
+Identifies a suspicious AutoIt process execution. Malware written as AutoIt script tend to rename the AutoIt executable
```
Comment (bm11100) [inline rules/windows/defense_evasion_masquerading_renamed_autoit.toml:9]:
```suggestion
Identifies a suspicious AutoIt process execution. Malware written as an AutoIt script tends to rename the AutoIt executable
```

Comment (bm11100) [inline rules/windows/defense_evasion_masquerading_renamed_autoit.toml:9]:
```suggestion
Identifies a suspicious AutoIt process execution. Malware written as an AutoIt script tends to rename the AutoIt executable
```

File: rules/windows/defense_evasion_rundll32_no_arguments.toml [hunk 9]
```diff
@@ -24,10 +23,10 @@ type = "eql"
 query = '''
 sequence with maxspan=1h
   [process where event.type in ("start", "process_started") and
-                                    /* uncomment once in winlogbeat */
-     (process.name : "rundll32.exe" /* or process.pe.original_file_name == "RUNDLL32.EXE" */ ) and
-     process.args_count < 2
+     (process.name : "rundll32.exe" or process.pe.original_file_name == "RUNDLL32.EXE") and
+      process.args_count < 2
   ] by process.entity_id
+
```
Comment (bm11100) [inline rules/windows/defense_evasion_rundll32_no_arguments.toml:29]: we dont need this empty line

File: rules/windows/defense_evasion_sdelete_like_filename_rename.toml [hunk 10]
```diff
@@ -12,18 +11,19 @@ file overwrite and rename operations.
...
 rule_id = "5aee924b-6ceb-4633-980e-1bde8cdb40c5"
 severity = "low"
 tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
-type = "query"
+type = "eql"
+
 
 query = '''
-event.category:file AND event.type:change AND file.name:/.+A+\.AAA/
+file where event.type == "change" and wildcard(file.name,"*AAA.AAA")
```
Comment (bm11100) [inline rules/windows/defense_evasion_sdelete_like_filename_rename.toml:26]: is the file name always uppercase? i.e., not case sensitive?

File: rules/windows/defense_evasion_suspicious_zoom_child_process.toml [hunk 11]
```diff
@@ -10,29 +9,21 @@ description = """
...
- not process.name:(Zoom.exe or WerFault.exe or airhost.exe or CptControl.exe or CptHost.exe or cpthost.exe or
-   CptInstall.exe or CptService.exe or Installer.exe or zCrashReport.exe or Zoom_launcher.exe or zTscoder.exe or
-   plugin_Launcher.exe or mDNSResponder.exe or zDevHelper.exe or APcptControl.exe or CrashSender*.exe or aomhost64.exe or
-   Magnify.exe or m_plugin_launcher.exe or com.zoom.us.zTranscode.exe or RoomConnector.exe or tabtip.exe or Explorer.exe or
-   chrome.exe or firefox.exe or iexplore.exe or outlook.exe or lync.exe or ApplicationFrameHost.exe or ZoomAirhostInstaller.exe or
-   narrator.exe or NVDA.exe or Magnify.exe or Outlook.exe or m_plugin_launcher.exe or mphost.exe or APcptControl.exe or winword.exe or
-   excel.exe or powerpnt.exe or ONENOTE.EXE or wpp.exe or debug_message.exe or zAssistant.exe or msiexec.exe or msedge.exe or
-   dwm.exe or vcredist_x86.exe or Controller.exe or Installer.exe or CptInstall.exe or Zoom_launcher.exe or ShellExperienceHost.exe or wps.exe)
+process where event.type in ("start", "process_started", "info") and
+ process.parent.name : "Zoom.exe" and process.name : ("cmd.exe", "powershell.exe")
```
Comment (bm11100) [inline rules/windows/defense_evasion_suspicious_zoom_child_process.toml:26]:
```suggestion
 process.parent.name : "Zoom.exe" and process.name : ("cmd.exe", "powershell.exe", "pwsh.exe")
```
has it been tested that cmd and powershell are the two likely children? any other script interpreters we want to account for?

File: rules/windows/defense_evasion_sdelete_like_filename_rename.toml (hunk 10 above)
Comment (Samirbous) [reply rules/windows/defense_evasion_sdelete_like_filename_rename.toml:26]: yes

**Review by devonakerr — APPROVED** [review]

**Review by bm11100 — APPROVED** [review]

**Pull request merged**